src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -113,15 +84,21 @@ public boolean isEmpty() {
         return size == 0;
     }
 
-    private void extendCapacity() {
-        int tempLength = arrayList.length;
-        T[] tempArray = (T[]) new Object[tempLength];
-        tempArray = arrayList;
-        arrayList = (T[]) new Object[(tempLength * 3) / 2 + 1];
-        System.arraycopy(tempArray, 0, arrayList, 0, size);
+    private void extendCapacityIfNecessary() {
+        if (size == arrayList.length) {
+            int tempLength = arrayList.length;
+            T[] tempArray = (T[]) new Object[tempLength];
+            tempArray = arrayList;
+            arrayList = (T[]) new Object[(tempLength * 3) / 2 + 1];
+            System.arraycopy(tempArray, 0, arrayList, 0, size);
+        }
     }
 
-    private boolean hasArraySpace() {
-        return size < arrayList.length;
+    private void checkIndexValidity(int index) {
+        if (index >= 0 && index < size) {
+            return;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index.");
+        }
```

Comment:
```suggestion
        if (index < 0 && index >= size) {
            throw new ArrayListIndexOutOfBoundsException("Wrong index.");
        }
```

Reply:
http://prntscr.com/1qvche9 - if I change this the logic will break

Reply:
done